contributors/design-proposals/storage/raw-block-pv.md

```diff
@@ -0,0 +1,739 @@
+
+Since the future of this in Kubernetes for GPUs and other plug-able devices is migrating to a device plugin architecture, there are 
+still differentiating components of storage that are enough to not to enforce alignment to their convention. Two factors when
+considering the usage of device plugins center around discoverability and topology of devices. Since neither of these are requirements
+for using raw block devices, the legacy method of populating the devices and appending it to the device array is sufficient.
+
+# Implementation Plan, Features & Milestones
+
+Phase 1: v1.9
+Feature: Pre-provisioned PVs to precreated devices 
+
+               Milestone 1: API changes
+
+               Milestone 2: Restricted Access 
+              
+               Milestone 3: Changes to the mounter interface as today it is assumed 'file' as the default.
+               
+               Milestone 4: Expose volumeMode to users via kubectl
+               
+               Milestone 5: PV controller binding changes for block devices
+               	       
+	       Milestone 6: Container Runtime changes
+	       
+	       Milestone 7: Initial Plugin changes (GCE, AWS & GlusterFS)
+	       
+	       Milestone 8: Disabling of provisioning where volumeMode == Block is not supported
+
+Phase 2:  v1.10
+Feature: Discovery of block devices 
+
+                Milestone 1: Dynamically provisioned PVs to dynamically allocated devices
+
+                Milestone 2: Privileged container concerns    
+		
+		Milestone 3: Flex volume update
+		
+Other considerations:
+
+               Reference volume driver change(Attach/Detach logic) for pre-provisioned PVs
+
+               Reference volume driver change(Provision/Delete logic) for dynamic provisioning
+
+# Plugin interface changes
+## New BlockVolume interface proposed design
+
+```
+// BlockVolume represents a directory used by pods or hosts on a node.
+// This interface provides methods to generate global map path and
+// pod device map path.
+type BlockVolume interface {
+	// GetGlobalMapPath returns a global map path which contains
+	// a symbolic links associated to a block device.
+	// ex. plugins/kubernetes.io/{PluginName}/{DefaultKubeletVolumeDevicesDirName}/{volumePluginDependentPath}/{pod uuid}
+	GetGlobalMapPath(spec *Spec) (string, error)
+	// GetPodDeviceMapPath returns a pod device map path
+	// and name of a symbolic link associated to a block device.
+	// ex. pods/{podUid}}/{DefaultKubeletVolumeDevicesDirName}/{escapeQualifiedPluginName}/{volumeName}
+	GetPodDeviceMapPath() (string, string)
+}
+```
+
+## New BlockVolumePlugin interface proposed design
+
+```
+// BlockVolumePlugin is an extend interface of VolumePlugin and is used for block volumes support.
+type BlockVolumePlugin interface {
+	VolumePlugin
+	// NewBlockVolumeMapper creates a new volume.BlockVolumeMapper from an API specification.
+	// Ownership of the spec pointer in *not* transferred.
```

Comment:
What do you mean by ownership, and not transferred?

Reply:
This comment is just copy and paste from VolumePlugin. will remove this line.